### PR TITLE
Add page object and test-suit for "Web Tables" page and inner "Registration Form" modal window

### PR DESCRIPTION
**Pull request by rybignat**
add registrationForm.page.ts
webTables.spec.ts(12 cases):
5 for "Web Tables" Page:

1. Check the functionality of the delete user button.
2. Verify selecting rows per page updates the table display.
3. Check the functionality of the search area.
4. Check the functionality of the buttons "Next" and "Previous" on page.
5. Check the functionality of navigating to a page by indicating its number.

7 for "Registration Form" modal window:

1. Open using "Add" button and close using "Close" button'.
2. Open using "Add" button and close using "Escape" keyboard button.
3. Create a new user and check if the data in web table has been transformed or lost.
4. Check the functionality of the error signal when entering incorrect data.
5. Check user data in the registration form window has not been transformed or lost.
6. Check the functionality of the ability to edit an existing user.
7. Check the functionality of protection against unwanted closing of the registration form.

update types.ts, README

### CODE REVIEW

File: tests/webTables.spec.ts [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+      await test.step('Click "Add" button', async () => {
+        await webTablesPage.clickAddButton()
+      })
+      await test.step('Check that "Registration Form" is visible', async () => {
+        const modalWindowState = await registrationFormPage.isModalWindowHeaderVisible()
+        expect(modalWindowState).toBeTruthy()
+      })
+      await test.step('Click on the cross in the upper right corner of the modal window', async () => {
+        await registrationFormPage.clickEscapeKeyboardButton()
+      })
```
Comment (AliaksandrPuchok) [inline tests/webTables.spec.ts]: Внимательнее

File: pageObjects/webTables.page.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  async deleteUserButtonClick (userEmail: string): Promise<void> {
+    const deleteUserButton = this.page.locator(`//div[@role="gridcell"][text()="${userEmail}"]/parent::div/descendant::span[@title="Delete"]`)
+    await deleteUserButton.click()
+  }
+
+  async searchInputFill (userEmail: string): Promise<void> {
+    await this.searchInput.fill(userEmail)
+  }
```
Comment (AliaksandrPuchok) [inline pageObjects/webTables.page.ts]: Поправь нейминг

File: Utils/Components/registrationForm.page.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  }
+
+  async isModalWindowHeaderVisible (): Promise<boolean> {
+    return await this.header.isVisible()
+  }
+
+  async checkForInputError (expectedColor: 'successColor' | 'errorColor', currentField: Locator): Promise<void> {
+    const color: string = expectedColor === 'errorColor' ? this.errorColor : this.successColor
+    await expect(currentField).toHaveCSS('border-color', color)
+  }
```
Comment (AliaksandrPuchok) [inline Utils/Components/registrationForm.page.ts]: надо поменять имя на checkInputValidationState или checkInputColorState или verifyInputState или checkInputState, потому что мы передаём как ошибку так и успешные варианты

**Review by AliaksandrPuchok — CHANGES_REQUESTED** [review]

File: tests/webTables.spec.ts (hunk 1 above)
Comment (rybignat) [reply tests/webTables.spec.ts]: Исправлено

File: pageObjects/webTables.page.ts (hunk 2 above)
Comment (rybignat) [reply pageObjects/webTables.page.ts]: Исправлено

File: Utils/Components/registrationForm.page.ts (hunk 3 above)
Comment (rybignat) [reply Utils/Components/registrationForm.page.ts]: Исправлено

**Review by AliaksandrPuchok — APPROVED** [review]

**Pull request merged**